model/result/ScoreWriterService.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA
+ */
+
+namespace oat\taoLtiConsumer\model\result;
+
+use common_exception_Error;
+use oat\oatbox\service\ConfigurableService;
+use oat\oatbox\service\exception\InvalidServiceManagerException;
+use oat\oatbox\service\ServiceManagerAwareTrait;
+use oat\taoDelivery\model\execution\DeliveryExecution;
+use oat\taoDelivery\model\execution\ServiceProxy;
+use oat\taoResultServer\models\classes\ResultServerService;
+use oat\taoResultServer\models\Exceptions\DuplicateVariableException;
+use taoResultServer_models_classes_OutcomeVariable as ResultServerOutcomeVariable;
+
+/**
+ * Class LtiXmlFormatterService
+ * Class to manage XML result data with score and to store it in DeliveryExecution
+ * @package oat\taoLtiConsumer\model\classes
+ */
+class ScoreWriterService extends ConfigurableService
+{
+    use ServiceManagerAwareTrait;
+
+    /**
+     * Store the score result into a delivery execution
+     * @param $result
+     * @return string
+     * @throws InvalidServiceManagerException
+     * @throws ResultException
+     * @throws common_exception_Error
+     * @throws DuplicateVariableException
+     */
+    public function store($result)
+    {
+        if (!(isset($result['score']) && $this->isScoreValid($result['score']))) {
```

Comment:
You can simplify the logic of this expression, given the boolean logic rules:
`! ( a && b)` => `!a || !b`
`! ( a || b)` => `!a && !b`